### PR TITLE
Fix GC ratio multiplier bug

### PR DESCRIPTION
**Pull request by mmottl**
This PR addresses the incorrect GC ratio multiplier as discussed in: https://github.com/ocaml/ocaml/issues/9073

### CODE REVIEW

**Review by gasche — APPROVED** [review]
This looks good to me. Note the change in behavior (which we believe fixes a bug) using `caml_custom_minor_ratio` instead of `caml_custom_major_ratio` to compute `max_minor`.

cc @damiendoligez, could you confirm that the fix is right?

(I think a Changes entry for this fix would be nice, in case people observe the bug in released versions and want to check whether it is fixed.)

**Comment by mmottl** [comment]
It seems `4.10` has already branched off.  Should I resubmit this bug fix including an updated `Changes` file for `4.10` before it is released?  It may be easier to then merge patches for `4.10` into `trunk`.  Not sure how urgent you think this fix is.

**Comment by gasche** [comment]
I think that the fix is not critical but important. I'm confident we can get a second review in time, and then I will take care of cherry-picking in the release branch. If nobody commented in a few days I'll go ahead and merge myself.

**Comment by mmottl** [comment]
I've updated the `4.10.0` section in `Changes` with the bugfix documentation.  I guess you'll just move it to the working section if the fix cannot make it into the `4.10.0` release anymore.

**Review by damiendoligez — APPROVED** [review]
Yes, the fix looks right, and there is time to apply it to 4.10.

**Comment by gasche** [comment]
I backported to 4.10 in 10d93692218739d69ca0f067950933271db806ad.

**Comment by mroch** [comment]
I suspect this bug is responsible for a very significant perf regression when we try to upgrade Flow to 4.08 or 4.09 (we're not ready for 4.10 yet due to `Marshal` changes). our workload involves many hundreds of thousands of `in_channel`s and we see time spent in GC roughly double.

Backporting this fix solves the regression. Any chance this could be backported to 4.08 and 4.09 like https://github.com/ocaml/ocaml/pull/10125 was? I'm happy to put up the PRs.

**Pull request merged**
